Update example apps to account for recent changes

sodic opened the pull request:
Updates the example apps with recent changes to the project structure and environment variable definitions. PRs covered by this:
- #753 
- #795 
- #683

shayneczyzewski commented on waspc/cli/src/Wasp/Cli/Command/Watch.hs:
```diff
@@ -40,7 +40,7 @@ watch waspProjectDir outDir ongoingCompilationResultMVar = FSN.withManager $ \mg
   chan <- newChan
   _ <- FSN.watchDirChan mgr (SP.fromAbsDir waspProjectDir) eventFilter chan
-  let watchProjectSubdirTree path = FSN.watchDirChan mgr (SP.fromAbsDir $ waspProjectDir </> path) eventFilter chan
+  let watchProjectSubdirTree path = FSN.watchTreeChan mgr (SP.fromAbsDir $ waspProjectDir </> path) eventFilter chan
```
Is this change supposed to be in this PR?

shayneczyzewski commented on waspc/run:
```diff
@@ -24,7 +24,7 @@ TEST_CMD="cabal test"
 TEST_CLI_CMD="cabal test cli-test"
 TEST_E2E_CMD="cabal test e2e-test"
-TEST_E2E_DELETE_GOLDENS="rm -r $PROJECT_ROOT/e2e-test/test-outputs/*-golden"
+TEST_E2E_DELETE_GOLDENS="rm -rf $PROJECT_ROOT/e2e-test/test-outputs/*-golden"
```
This change may be old now too.

shayneczyzewski commented on waspc/new-release:
```diff
@@ -6,71 +6,71 @@
-    let newContents = beforeMatch ++ matchBeforeVersion ++ newVersion ++ afterMatch
-    return (newContents, oldVersion)
+  let (beforeMatch, match, afterMatch, submatches) =
+        cabalFileContents TR.=~ ("^(version: *)([0-9]+\\.[0-9]+\\.[0-9]+)" :: String) :: (String, String, String, [String])
```
Looks like on this line we lost an `\\.[0-9]+` at the end. Was that intended?

shayneczyzewski commented on examples/thoughts/.env.server:
```diff
@@ -0,0 +1 @@
+DATABASE_URL=postgresql://postgres:devpass1234@localhost:5432/thoughts-dev
```
We may want to remove this file entirely since it is `.gitignore`'d and the only env file we have committed for examples.

sodic replied:
Yeah, thought so too. I can remove it.

shayneczyzewski reviewed (commented):
Looking good @sodic! Tried running a few, all seems good. Nice work refactoring all of them. Left a few comments/questions 👍🏻 I wonder if, given your updates here, if in the `ChangeLog` update describing this we can give a TL;DR update of what they can do to refactor their apps? I know it is pretty straightforward but may help some if we list it out.

sodic commented on examples/tutorials/ItWaspsOnMyMachine/migrations/20220819110846_init/migration.sql:
```diff
@@ -1,5 +0,0 @@
--- CreateTable
-CREATE TABLE "Excuse" (
```
This file was causing DB errors. We have two init migrations, this was the one that was out of date. I kept the other one.

sodic commented:
@shayneczyzewski Addressed all your comments and added the migration guide to the changelog. I think this is it.

**Note:** I didn't update this page: https://wasp-lang.dev/blog/2021/03/02/wasp-alpha. It's pretty old and hasn't been updated in ages (e.g., it uses Wasp syntax I've never seen). Everything else should be taken care of.

shayneczyzewski commented on waspc/ChangeLog.md:
```diff
@@ -7,6 +7,110 @@
 - Updates required Node LTS version from version 16 to version 18. This Node ecosystem change happened on 2022-10-25: https://github.com/nodejs/Release
 
+#### Significant changes to Wasp project structure
+This was the file tree of a newly generated in the previous version of Wasp
```
```suggestion
This was the file tree of a newly generated project in the previous version of Wasp
```

shayneczyzewski commented on waspc/ChangeLog.md:
````diff
@@ -7,6 +7,110 @@
+     ```
+     Do this for all external imports in your `.wasp` file. After you're done, there shouldn't be any occurences of the string `"@ext"`.
+     
+That's it! You should now have a fully working Wasp project in the `foo` directory.
````
Awesome guide!!! 🥳

sodic replied:
Thanks, I accumulated a lot of experience with this in recent days :)

shayneczyzewski reviewed (approved):
Awesome job @sodic! Found one minor typo but all this looks great. Love the Changelog update! 🚀 great job getting all these examples up to date 👍🏻
